src/datagen/main/java/mekanism/common/integration/computer/ComputerHelpProvider.java

```diff
@@ -0,0 +1,309 @@
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+import java.util.concurrent.CompletableFuture;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import mekanism.api.annotations.ParametersAreNotNullByDefault;
+import mekanism.common.Mekanism;
+import mekanism.common.MekanismDataGenerator;
+import mekanism.common.MekanismDataGenerator.IOConsumer;
+import mekanism.common.content.filter.IFilter;
+import mekanism.common.content.filter.IItemStackFilter;
+import mekanism.common.content.filter.IModIDFilter;
+import mekanism.common.content.filter.ITagFilter;
+import mekanism.common.content.miner.MinerFilter;
+import mekanism.common.content.oredictionificator.OredictionificatorFilter;
+import mekanism.common.content.oredictionificator.OredictionificatorItemFilter;
+import mekanism.common.content.qio.filter.QIOFilter;
+import mekanism.common.content.transporter.SorterFilter;
+import mekanism.common.integration.computer.MethodHelpData.Param;
+import mekanism.common.integration.computer.MethodHelpData.Returns;
+import mekanism.common.lib.Version;
+import mekanism.common.lib.multiblock.MultiblockData;
+import mekanism.common.tile.base.TileEntityMekanism;
+import mekanism.common.tile.factory.TileEntityFactory;
+import mekanism.common.tile.factory.TileEntityItemStackGasToItemStackFactory;
+import mekanism.common.tile.laser.TileEntityBasicLaser;
+import mekanism.common.tile.multiblock.TileEntityInductionPort;
+import mekanism.common.tile.prefab.TileEntityAdvancedElectricMachine;
+import mekanism.common.tile.prefab.TileEntityMultiblock;
+import mekanism.common.tile.prefab.TileEntityProgressMachine;
+import mekanism.common.tile.qio.TileEntityQIOComponent;
+import mekanism.common.tile.qio.TileEntityQIOFilterHandler;
+import mekanism.common.util.MekCodecs;
+import net.minecraft.MethodsReturnNonnullByDefault;
+import net.minecraft.Util;
+import net.minecraft.data.CachedOutput;
+import net.minecraft.data.DataProvider;
+import net.minecraft.data.PackOutput;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.util.CsvOutput;
+import net.minecraft.util.CsvOutput.Builder;
+import net.minecraft.util.ExtraCodecs;
+import org.jetbrains.annotations.NotNull;
+import org.yaml.snakeyaml.DumperOptions;
+import org.yaml.snakeyaml.nodes.Node;
+
+@MethodsReturnNonnullByDefault
+@ParametersAreNotNullByDefault
+public class ComputerHelpProvider implements DataProvider {
+    private static final String[] METHOD_CSV_HEADERS = {"Class", "Method Name", "Params", "Returns", "Restriction", "Requires Public Security", "Description"};
+    private static final String[] ENUM_CSV_HEADERS = {"Type Name", "Values"};
+    private final PackOutput.PathProvider pathProvider;
+    private final String modid;
+
+    public ComputerHelpProvider(PackOutput output, String modid) {
```

Comment:
Does this include mekanism generators' data and that is why this provider is never called with mekanism generator's modid?

Reply:
yes. Since it's 99% for website documentation where they wouldn't be separated I didn't bother splitting them

Reply:
No idea how large a file it is so it may not matter, but do we potentially want to be excluding it from the built jar then?

Reply:
I think it's good to have in there somewhere as a definitive reference, at least the markdown/yaml file. CSV was just to add to the google sheet

It should compress down to very little, as the strings would share zip dictionary entries with the class files as they would contain all the strings

Reply:
Sounds reasonable enough to me